TangoRosStreamer/app/src/main/java/eu/intermodalics/tango_ros_streamer/activities/RunningActivity.java

```diff
@@ -364,6 +383,32 @@ public void onClick(View view) {
         }
     }
 
+    public void onClickOkLoadNavMapDialog(final String mapName) {
+        if (mapName == null || mapName.isEmpty()) {
+            Log.e(TAG, "Map name is null or empty, unable to load the navigation map");
+            displayToastMessage(R.string.map_name_error);
+            return;
+        }
+        new AsyncTask<Void, Void, Void>() {
+            @Override
+            protected Void doInBackground(Void... params) {
+                mTangoServiceClientNode.callLoadNavMapService(mapName);
+                return null;
+            }
+        }.execute();
+    }
+
+    @Override
+    public void onLoadNavMapServiceCallFinish(boolean success, final String message) {
+        if (success) {
+            displayToastMessage(R.string.load_nav_map_success);
+            Log.i(TAG, message);
+        } else {
+            Log.e(TAG, "Error while loading navigation map: " + message);
+            displayToastMessage(R.string.load_nav_map_error);
```

Comment:
A toast message is only displayed on the screen for a very short time. It is therefore easy to miss. Does it make sense to show the load map dialog again when the map failed to load with extra message in red text, e.g. Map "map name" failed to load for some reason, try again with a different map<